src/helics/shared_api_library/MessageFederateExport.cpp

```diff
@@ -702,7 +702,7 @@ const char* helicsMessageGetString(helics_message message)
     if (mess == nullptr) {
         return nullcstr;
     }
-    return mess->data.data();
+    return mess->data.to_string().data();
```

Comment:
While technically correct, the cast-chain triggered by this `to_string().data()` goes `byte*`->`const char*` (the target) -> construction of `std::string_view` -> access of `const char*` data member. 
It may be worth considering simply adding a `const char* data()` member to `SmallBuffer` to reduce this cast chain in situations where there's no actual need for the string view to be constructed.